modal/_utils/grpc_utils.py

```diff
@@ -256,19 +257,24 @@ async def retry_transient_errors(
                 # TODO: update to newer version (>=0.4.8) once stable
                 raise exc
 
-            logger.debug(f"Retryable failure {repr(exc)} {n_retries=} {delay=} for {fn.name} ({idempotency_key[:8]})")
-
-            n_retries += 1
+            logger.debug(
+                f"Retryable failure {repr(exc)} {n_total_retries=} {delay=} for {fn.name} ({idempotency_key[:8]})"
+            )
 
             if (
                 retry_warning_message
-                and n_retries % retry_warning_message.warning_interval == 0
+                and n_total_retries % retry_warning_message.warning_interval == 0
```

Comment:
Note that neither `n_total_retries` nor `n_limited_retries` is exactly right here since the warning logic should warn about a _separate_ set of specific error types!